src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -2986,6 +2986,57 @@ public final T blockLastMillis(long timeout) {
 	}
 
 	/**
+	 * Triggers side-effects when the {@link Flux} emits an item, fails with an error or
+	 * completes successfully. This is provided as convenience to avoid using the
+	 * three corresponding {@code doOn} methods individually. Note that at least one
+	 * callback is required.
+	 *
+	 * @param nextConsumer the callback to call on {@link Subscriber#onNext(Object)}
+	 * @param errorConsumer the callback to call on {@link Subscriber#onError(Throwable)}
+	 * @param completeRunnable the callback to call on {@link Subscriber#onComplete()}
+	 * @return an observed {@link Flux}
+	 * @see #doOnNext(Consumer)
+	 * @see #doOnError(Consumer)
+	 * @see #doOnComplete(Runnable)
+	 */
+	public final Flux<T> doOnEach(Consumer<? super T> nextConsumer,
+			Consumer<? super Throwable> errorConsumer,
+			Runnable completeRunnable) {
```

Comment:
IMO:
- This isn't 100% required, using consecutive doOnNext() and doOnError() isn't particularly much more verbose.
- If this method does exist i would either, limit it to next/error consumer only, or expand it to include afterTerminate runnable too so it has al signals.

In order to include all signals this should include a terminateRunnable also shouldn't it? 

I don't see huge value of this over requiring doOnNext(), doOnComplete